crates/ursa-index-provider/src/advertisement.rs

```diff
@@ -22,6 +22,8 @@ struct Metadata {
     // Data is specific to the identified protocol, and provides data, or a
     // link to data, necessary for retrieval.
     Data: Vec<u8>,
```

Comment:
If I'm reading the docs correctly, this is for protocol (bitswap, graphsync, etc) specific stuff but we are just adding this extra stuff (FleekNetwork). Do you think it will be ok to add this other data for the file size? @theBeardA

Reply:
with bitswap, it is ok since data is empty. And I just realized that with graphsync, we might need to more information to Data field.
but on second thought, the data field is just vec<u8>, so we can have any kind of data encoded. and then when we retrieve, we should know what we need to do with this data and what is the expected result from decoding this data e.g GraphSyncStruct and other fields we are adding

Reply:
So I'm thinking something like 
```rust
struct Metadata {
  ProtocolID: u128,
  Size: u64,
  Data: Vec<u8>,
}
```

Simple enough for now. We can then serialize/deserialize Data for protocol information. The indexer does not validate metadata right? If so, I think we will be OK to move forward then.

Reply:
Another question I have is, `Size` makes sense if we are querying a file so `Size` is the file size but what if we are querying a block, do we currently support `put`ting a single block as apposed to an entire file?

Reply:
cc @scott-dn @theBeardA

Reply:
Yes, we can add that, but in order to increase the order, it's expected to follow as below:
![image](https://user-images.githubusercontent.com/26348060/210930129-48a081ef-e8fe-4577-b3fc-2058d07fcecc.png)

Reply:
@scott-dn what about `Size`? Can we expect to have `Size` for any CID? Considering that CID can be CID of root for a file or just a CID for some block (not the entire file).

Reply:
I don't think there are any problems with that. We're able to know the size when storing a car file on the cache node.

Reply:
Ok, cool 😊 I just wanted to make sure I understood the current use cases. I'll keep moving forward with this PR.